src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +257,68 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+  private void flushMessageBuffer() {
+    int batchSize = config.getBatchSize();
+    String jsonBatchKey = config.getJsonBatchKey();
+    boolean useJsonBatchKey = !Strings.isNullOrEmpty(jsonBatchKey) && batchSize > 1;
+      switch (config.getMessageFormat()) {
+          case "JSON":
+              if (useJsonBatchKey) {
```

Comment:
make it like 
if(batchsize == 1) {
normal append to messages}
else {
if(useJsonBatchKey) {
String.format("{ "%s" : [%s]}");
}else{
String.format("[%s]");
}
}
it will be more readable, create variables for above values.

Reply:
String.format is a better way to write clean code.
This comes with a slight cost of performance. 


Updated code
```java
case "JSON":
  String messageBody;
  if (useJsonBatchKey) {
    messageBody = String.format("{\"%s\":[%s]}", jsonBatchKey, String.join(",", messageBuffer));
  } else {
    messageBody = batchSize == 1 ?
        String.format("%s", String.join(",", messageBuffer)) :
        String.format("[%s]", String.join(",", messageBuffer));
  }
  messages.append(messageBody);
  break;
```

Reply:
in case of batch size ==1 remove unnecessary calling of format and join method.
In case of a small string like this, it is better to use format as performance impact will be very minimal

Reply:
Removed unnecessary calling of format and join method 
```java
messageBody = batchSize == 1 ? messageBuffer.get(0) :
    String.format("[%s]", String.join(",", messageBuffer));
```